base/files/file_enumerator_starboard.cc

```diff
@@ -133,15 +131,25 @@ std::vector<FileEnumerator::FileInfo> FileEnumerator::ReadDirectory(
   };
 
   std::vector<FileEnumerator::FileInfo> ret;
-  // We test if SbDirectoryGetNext returns the parent directory, i.e. |..|,
+  // We test if readdir_r returns the parent directory, i.e. |..|,
   // because whether or not it is returned is platform-dependent and we need to
   // be able to guarantee it is returned when the INCLUDE_DOT_DOT bitflag is
   // set.
   bool found_dot_dot = false;
 
   std::vector<char> entry(kSbFileMaxName);
+  struct dirent dirent_buffer;
```

Comment:
No need for action or changes in this PR, but just leaving a comment.

We should probably aim to entirely `_delete_` `file_enumerator_starboard.cc` and simply use `file_enumerator_posix.cc` wholesale, without any Starboard #ifdefs . To that end, i would reference how the loop is written in that module and probably track it a little closer here.

But - it's fine to get this in as is, if tests are passing. Just a call-out that we seem to have substantially different behavior still.